app/src/main/java/com/example/bigowlapp/utils/AlarmBroadcastReceiverManager.java

```diff
@@ -0,0 +1,120 @@
+        getSchedulesForUser(userID)
+                .addOnSuccessListener(scheduleList -> {
+                    for (Schedule schedule : scheduleList) {
+                        Intent intent = new Intent(context, AlarmBroadcastReceiver.class);
+                        intent.putExtra(EXTRA_UID, schedule.getUid());
+                        intent.putExtra(EXTRA_LATITUDE, schedule.getLocation().getLatitude());
+                        intent.putExtra(EXTRA_LONGITUDE, schedule.getLocation().getLongitude());
+                        PendingIntent pendingIntent = PendingIntent
+                                .getBroadcast(context, requestCode, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+                        alarmManager.set(AlarmManager.RTC_WAKEUP,
+                                schedule.getStartTime().toDate().getTime(), pendingIntent);
+                    }
+                }).addOnFailureListener(error -> Toast.makeText(context, error.getMessage(), Toast.LENGTH_SHORT).show());
+    }
+
+    /**
+     * Obtains the list of schedules for the user, and filters in schedules that have been accepted
+     *
+     * @param userID The Id of the user
+     * @return A Task that contains a list of schedule for the user that hasn't been attended
+     */
+    private Task<List<Schedule>> getSchedulesForUser(String userID) {
+
+        // TODO: --START-- REMOVE BEFORE MERGING PULL REQUEST. THIS IS ONLY FOR TESTING PURPOSES.
+        if (userID.equals("testing")) {
+            List<Schedule> acceptedScheduleList = new ArrayList<>();
+
+            Schedule schedule = new Schedule();
+            schedule.setUid("Hello there");
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTime(new Date());
+            calendar.add(Calendar.SECOND, 30);
+            Timestamp timestamp = new Timestamp(calendar.getTime());
+            schedule.setStartTime(timestamp);
+            GeoPoint geoPoint = new GeoPoint(10, -82);
+            schedule.setLocation(geoPoint);
+
+            acceptedScheduleList.add(schedule);
+
+            return Tasks.forResult(acceptedScheduleList);
+        }
+        // TODO: --END-- REMOVE BEFORE MERGING PULL REQUEST. THIS IS ONLY FOR TESTING PURPOSES.
+
+        return scheduleRepository.getTaskListSchedulesForUser(userID)
+                .continueWithTask(task -> {
+                    if (task.isSuccessful()) {
+                        QuerySnapshot tDocs = task.getResult();
+                        List<Schedule> scheduleList = tDocs.toObjects(Schedule.class);
+                        List<Schedule> acceptedScheduleList = new ArrayList<>();
+                        for (Schedule schedule : scheduleList) {
+                            if (schedule.getUserScheduleResponseMap()
+                                    .get(userID).getResponse() == Response.ACCEPT) {
+                                acceptedScheduleList.add(schedule);
+                            }
+                        }
```

Comment:
This can be refactored using [Replace Loop with Pipeline](https://refactoring.com/catalog/replaceLoopWithPipeline.html) using java 8 stream.

Although I think this is ok too tbh.

Reply:
Although, the performance will be worst according to this source (https://stackoverflow.com/questions/44180101/in-java-what-are-the-advantages-of-streams-over-loops), it's nice for readability and I don't think performance will take a significant hit, so I'll do the refactor.

Reply:
I actually didn't know that, good to know.